src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] values;
+    private int size;
+
+    public ArrayList() {
+        values = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
+        values[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        grow();
+        if (index != size) {
+            checkIndexWithSameSize(index);
+            System.arraycopy(values, index, values, index + 1, size - index);
+        }
+        values[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        T[] a = (T[]) new Object[list.size()];
+        for (int i = 0; i < list.size(); i++) {
+            a[i] = list.get(i);
+        }
+        int varLength = a.length;
+        grow();
+        System.arraycopy(a,0,values,size, varLength);
+        size += varLength;
```

Comment:
A lot of redundant code. Let's simplify it. I'd recommend iterating throughout a list and add new elements using your add() method